torch/csrc/jit/passes/freeze_module.h

```diff
@@ -0,0 +1,23 @@
+/** \brief This file defines passes to freeze pytorch module..
+ *
+ * The passes have python-bindings and can be invoked directly or as a part of
+ * general optimization pipeline (details TBD).
+ */
+#pragma once
+
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/script/module.h>
+
+/** \brief Freeze Module, i.e., Assume all atrribute are constants.
+ *
+ * constant propagate attribute and optimize module attributes agressively
+ * inline into forward to enable optimizations.
+ */
+
+namespace torch {
+namespace jit {
+
+TORCH_API script::Module FreezeModule(const script::Module& module);
```

Comment:
I think this should take a list of methods to preserve rather than just being forward. Things like SetAttr checking have to occur across all methods, not just a single method, so it is not as simple as looping over each method, however, in total it is not a large change.

A common issue with APIs on models is to only support a single forward method, but this causes a bunch of weird downstream problems (caffe2 init_net and predict_net are classic examples of this issue). We've also had to modify tracing to have multiple methods as well. Without this support people we submitting issues for things like tracing constants, and it took us forever to realize that what people needed was to export multiple methods at the same time.